### PR TITLE
Enhance Test Coverage for `@yamada-ui/calendar`

### LINKED ISSUE #1025: Enhance Test Coverage for `@yamada-ui/calendar`
package: @yamada-ui/calendar

### Description
`@yamada-ui/calendar` has a test coverage of less than 90%. We will enhance it to above 90%.

### Target files
- [packages/components/calendar/src/calendar-utils.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts)
  - [L144](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L144)
  - [L171](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L171)
  - [L174](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L174)
  - [L212](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L212)
  - [L218](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L218)
  - [L277](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L277)
  - [L287](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L287)
  - [L295](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L295)
  - [L303](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/calendar-utils.ts#L303)
- [packages/components/calendar/src/use-calendar-header.tsx](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx)
  - [L40](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx#L40)
  - [L50](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx#L50)
  - [L57](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx#L57)
  - [L62](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx#L62)
  - [L77](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx#L77)
  - [L82](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx#L82)
  - [L100](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-header.tsx#L100)
- [packages/components/calendar/src/use-calendar-picker.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts)
  - [L240](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts#L240)
  - [L282](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts#L282)
  - [L290](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts#L290)
  - [L298](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts#L298)
  - [L303](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts#L303)
  - [L316](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts#L316)
  - [L371](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-calendar-picker.ts#L371)
- [packages/components/calendar/src/use-month-list.tsx](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx)
  - [L47](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx#L47)
  - [L69](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx#L69)
  - [L91](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx#L91)
  - [L136](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx#L136)
  - [L149](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx#L149)
  - [L158](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx#L158)
  - [L194](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month-list.tsx#L194)
- [packages/components/calendar/src/use-month.tsx](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month.tsx)
  - [L98](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month.tsx#L98)
  - [L138](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month.tsx#L138)
  - [L178](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month.tsx#L178)
  - [L312](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month.tsx#L312)
  - [L319](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-month.tsx#L319)
- [packages/components/calendar/src/use-multi-date-picker.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts)
  - [L100](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L100)
  - [L104](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L104)
  - [L109](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L109)
  - [L129](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L129)
  - [L140](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L140)
  - [L146](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L146)
  - [L160](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L160)
  - [L164](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-multi-date-picker.ts#L164)
- [packages/components/calendar/src/use-range-date-picker.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts)
  - [L67](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L67)
  - [L73](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L73)
  - [L112](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L112)
  - [L121](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L121)
  - [L142](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L142)
  - [L169](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L169)
  - [L200](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L200)
  - [L225](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L225)
  - [L272](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L272)
  - [L280](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L280)
  - [L321](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L321)
  - [L329](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-range-date-picker.ts#L329)
- [packages/components/calendar/src/use-year-list.tsx](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx)
  - [L40](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L40)
  - [L62](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L62)
  - [L84](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L84)
  - [L112](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L112)
  - [L129](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L129)
  - [L140](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L140)
  - [L178](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L178)
  - [L180](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-list.tsx#L180)
- [packages/components/calendar/src/use-year-picker.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-picker.ts)
  - [L48](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/use-year-picker.ts#L48)
- [packages/components/calendar/src/year-picker.tsx](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/year-picker.tsx)
  - [L73](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/calendar/src/year-picker.tsx#L73)

### Problem Statement/Justification
Testing is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

### PR DESCRIPTION
**Pull request by illionillion**
Closes #1025  

## Description

Enhance Test Coverage for `@yamada-ui/calendar`

- split test code
- add new test code

### CODE REVIEW

File: packages/components/calendar/tests/date-picker.test.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      />,
+    )
+
+    const popover = container.querySelector(".ui-popover")
+
+    act(() => {
+      fireEvent.focus(screen.getByPlaceholderText("basic"))
+    })
+
+    expect(popover).toHaveStyle("visibility: visible;")
```
Comment (108yen) [inline packages/components/calendar/tests/date-picker.test.tsx:45]:
```ts
expect(popover).toHaveStyle({visibility:"visible"})
```

**Review by 108yen — CHANGES_REQUESTED** [review]
remove `act`
```ts
 act(() => {
      fireEvent.xxx
    })
```

File: packages/components/calendar/tests/date-picker.test.tsx (hunk 1 above)
Comment (illionillion) [reply packages/components/calendar/tests/date-picker.test.tsx]: resolved

**Comment by 108yen** [comment]
@illionillion 
unnecessary `render`
```diff ts
-const { container } = render(<Component />)
-await a11y(container)
+await a11y(<Component />)
```

**Comment by illionillion** [comment]
> @illionillion unnecessary `render`
> 
> ```diff
> -const { container } = render(<Component />)
> -await a11y(container)
> +await a11y(<Component />)
> ```

When I do that, I get this error.

image

**Comment by 108yen** [comment]
@illionillion 

I think it's a bug too.

**Comment by illionillion** [comment]
> @illionillion
> 
> I think it's a bug.

Do you want to merge and create a new issue?

**Comment by 108yen** [comment]
@illionillion 
I hand it over to you.

**Comment by koralle** [comment]
@illionillion 

When using real timers in the test code, the test results may vary each time the test is run.   
Call `vi.useFakeTimers()` at the beginning of your test to mock the timers, and call `vi.useRealTimers()` at the end to restore the real timers.

See: 
* https://vitest.dev/api/vi.html#vi-usefaketimers
* https://qiita.com/mori_goq/items/122d5eb4cafd903cdeb9
* https://zenn.dev/bs_kansai/articles/21962b63727f27

**Comment by illionillion** [comment]
> @illionillion
> 
> When using real timers in the test code, the test results may vary each time the test is run. Call `vi.useFakeTimers()` at the beginning of your test to mock the timers, and call `vi.useRealTimers()` at the end to restore the real timers.
> 
> See:
> 
> * https://vitest.dev/api/vi.html#vi-usefaketimers
> * https://qiita.com/mori_goq/items/122d5eb4cafd903cdeb9
> * https://zenn.dev/bs_kansai/articles/21962b63727f27

Is this correct?

https://github.com/yamada-ui/yamada-ui/pull/1707/commits/8a4d3c96b09b3d766a3963f8f326eb09b28709a4

**Comment by illionillion** [comment]
> @illionillion I hand it over to you.

I resolved this.

**Review by 108yen — DISMISSED** [review]

**Comment by illionillion** [comment]
@108yen @koralle Please one more review.

**Review by 108yen — APPROVED** [review]

**Pull request merged**